codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -599,14 +601,28 @@ void resizeIfNecessary(int index) {
                 }
             }
         }
+
+        @Override
+        public boolean equals(final Object obj) {
+            return super.equals(obj);
+        }
+
+        @Override
+        public int hashCode() {
+            long value = identity;
+            if (value == 0) {
+                return System.identityHashCode(this);
```

Comment:
Within a single TCP connection, all HTTP/2 streams are unique and this change should not cause any problems. Only if it happens that users compare hashCode of streams from different connections, then the hashCode can be identical, but equals will still return false. I don't have a use-case in mind for that, just brainstorming. So, maybe we should not overthink.